ci/new_tsqa/tests/test_tls_ticket_key_rotation.py

```diff
@@ -0,0 +1,175 @@
+
+        # check whether TLS session tickets are received by s_client.
+        stdout, _ = tsqa.utils.run_sync_command(ticket_cmd, stdout=subprocess.PIPE, shell=True)
+        ticket_exists = False
+        for line in stdout.splitlines():
+            text = line.strip()
+            if text.startswith("TLS session ticket:"):
+                ticket_exists = True
+                break
+        self.assertTrue(ticket_exists)
+
+        # check whether the session has been reused
+        reused = False
+        ticket_cmd = 'echo | openssl s_client -connect {0}:{1} -sess_in {2}'.format(addr[0], addr[1], sess);
+        stdout, _ = tsqa.utils.run_sync_command(ticket_cmd, stdout=subprocess.PIPE, shell=True)
+        for line in stdout.splitlines():
+            text = line.strip()
+            if text.startswith("Reused, TLSv1/SSLv3,"):
+                reused = True
+                break
+        self.assertTrue(reused)
+
+        # negative test case. The session is not reused.
+        reused = False
+        ticket_cmd = 'echo | openssl s_client -connect {0}:{1}'.format(addr[0], addr[1]);
+        stdout, _ = tsqa.utils.run_sync_command(ticket_cmd, stdout=subprocess.PIPE, shell=True)
+        for line in stdout.splitlines():
+            text = line.strip()
+            if text.startswith("Reused, TLSv1/SSLv3,"):
+                reused = True
+                break
+        self.assertFalse(reused)
+
+    def test_tls_ticket_rotation(self):
+        '''
+        Make sure the new ticket key is loaded
+        '''
+        addr = ('127.0.0.1', self.ssl_port)
+        self._get_cert(addr)
+
+        '''
+        openssl s_client -connect server_ip:ssl_port -tls1 < /dev/null
+        '''
+
+        # Generate and push a new ticket key
+        rotate_cmd = 'openssl rand 48 -base64 > {0}'.format(helpers.tests_file_path('rsa_keys/ssl_ticket.key'))
+        stdout, _ = tsqa.utils.run_sync_command(rotate_cmd, stdout=subprocess.PIPE, shell=True)
+
+        # touch the ssl_multicert.config file
+        ssl_multicert = os.path.join(self.environment.layout.sysconfdir, 'ssl_multicert.config')
+
+        read_renewed_cmd = os.path.join(self.environment.layout.bindir, 'traffic_line') + ' -r proxy.process.ssl.total_ticket_keys_renewed'
+
+        # Check whether the config file exists.
+        self.assertTrue(os.path.isfile(ssl_multicert), ssl_multicert)
+        touch_cmd = which('touch') + ' ' +  ssl_multicert
+        tsqa.utils.run_sync_command(touch_cmd, stdout=subprocess.PIPE, shell=True)
+
+        count = 0
+        while True:
+            try:
+                stdout, _ = tsqa.utils.run_sync_command(read_renewed_cmd, stdout=subprocess.PIPE, shell=True)
+                old_renewed = stdout
+                break
+            except Exception:
+                ++count
+                # If we have waited more than 30 seconds and the command still failed, quit here.
+                if count > 30:
+                    self.assertTrue(False)
```

Comment:
You can define an error message as the second ard to this. That way if this fails the person running the tests can have a message instead of an assert True != False (https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertTrue)